src/librustc_lint/types.rs

```diff
@@ -364,6 +364,7 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for TypeLimits {
             }
         }
 
+        #[cfg_attr(not(stage0), allow(usage_of_ty_tykind))]
```

Comment:
This is required, since there is a `TyKind` built here:
https://github.com/rust-lang/rust/blob/e208034e0992bfb640978f139a1e0f2e23ae5f0c/src/librustc_lint/types.rs#L107
and here:
https://github.com/rust-lang/rust/blob/e208034e0992bfb640978f139a1e0f2e23ae5f0c/src/librustc_lint/types.rs#L162
and AFAIK you can't build a full `TyS` from scratch.

Reply:
Since this only uses `Int` and `Uint`, make the argument `attr::IntType::UnsignedInt` directly

Reply:
Ah nice. We even get an exhaustive match with this change and can remove a `bug!` macro call.